### PR TITLE
Feature detection dataset

### PR DESCRIPTION
**Pull request by PososikTeam**

### CODE REVIEW

File: torchok/data/datasets/detection/detection.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        self.bbox_column = bbox_column
+        self.bbox_dtype = bbox_dtype
+
+        self.bbox_format = bbox_format
+
+        if annotation_path.endswith('csv'):
+            self.df = pd.read_csv(self.data_folder / annotation_path)
+            self.df[self.bbox_column] = self.df[self.bbox_column].apply(json.loads)
+            self.df[self.target_column] = self.df[self.target_column].apply(json.loads)
+        else:
```
Comment (Animatory) [inline torchok/data/datasets/detection/detection.py:96]: `elif annotation_path.endswith('pkl')`
else: raise error

File: torchok/data/datasets/detection/detection.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    def get_raw(self, idx: int) -> dict:
+        record = self.df.iloc[idx]
+        image_path = self.data_folder / record[self.input_column]
+        sample = {'image': self._read_image(image_path), 'index': idx}
+
+        if not self.test_mode:
+            target = record[self.target_column]
+            bboxes = record[self.bbox_column]
+            sample['label'] = target
```
Comment (Animatory) [inline torchok/data/datasets/detection/detection.py:132]: `sample['label'] = record[self.target_column]`
`sample['bboxes'] = record[self.bbox_column]`

File: tests/data/datasets/test_detection.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    def test_shape_when_grayscale(self):
+        super().test_shape_when_grayscale()
+
+    def test_input_dtype(self):
+        super().test_input_dtype()
+
+    def test_output_format(self):
+        ds = self.create_dataset()
+        self.assertListEqual(list(ds[0].keys()), ['image', 'index', 'label', 'bboxes'])
```
Comment (Animatory) [inline tests/data/datasets/test_detection.py:34]: Move  `['image', 'index', 'label', 'bboxes']` in setUp

File: torchok/data/datasets/detection/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from torchok.data.datasets.detection.detection import DetectionDataset
```
Comment (Animatory) [inline torchok/data/datasets/detection/__init__.py:1]: `import torchok.data.datasets.detection.detection`

File: tests/data/datasets/test_detection.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        super().test_shape_when_transformed()
+
+    def test_shape_when_grayscale(self):
+        super().test_shape_when_grayscale()
+
+    def test_input_dtype(self):
+        super().test_input_dtype()
+
+    def test_output_format(self):
+        ds = self.create_dataset()
```
Comment (Animatory) [inline tests/data/datasets/test_detection.py:34]: Call from super()

**Review by Animatory — APPROVED** [review]
